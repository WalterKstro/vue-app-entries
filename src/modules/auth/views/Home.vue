<template>
  <h1>Home Sidebar</h1>
  <button class="btn btn-primary" @click="onClick">Primary</button>
</template>

<script>
export default {
  methods: {
    onClick() {
      this.$router.push({name: 'entry-no-selected'});
    }
  }
}
</script>

<style>

</style>