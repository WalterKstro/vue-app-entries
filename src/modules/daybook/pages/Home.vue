<template>
  <Navbar/>
  <div class="container">
    <main class="row wrapper__daybook">
      <div class="col-12 col-md-4">
        <EntryesList/>
      </div>
      <div class="col-12 col-md-8 d-flex flex-column justify-content-center">
        <router-view></router-view>
      </div>
    </main>
  </div>
</template>

<script>
import { defineAsyncComponent } from "vue"

export default {
  components: {
    Navbar: defineAsyncComponent(() => import("../components/Navbar.vue")),
    EntryesList: defineAsyncComponent(() => import("../components/EntryList.vue")),
  },
}
</script>

<style lang="scss" scoped>
.wrapper__daybook{
  height: calc(100vh - 72px);
}
</style>