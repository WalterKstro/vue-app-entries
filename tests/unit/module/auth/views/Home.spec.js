import { shallowMount } from '@vue/test-utils';
import Home from '../../../../../src/modules/auth/views/Home.vue';

describe('Pruebas en la vista home del modulo auth', () => {

    test('Debe de renderizar correctamente el componente View', () => {
        const wrapper = shallowMount(Home);
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Debe de llamarse el mÃ©todo push, del router', () => {
        const objectRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: objectRouter
                }
            }
        });
        
        wrapper.find('button').trigger('click');

        expect(objectRouter.push).toHaveBeenCalled();
        expect(objectRouter.push).toHaveBeenCalledWith({name: 'entry-no-selected'});
    });
})