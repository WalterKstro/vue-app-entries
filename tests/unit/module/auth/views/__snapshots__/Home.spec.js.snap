// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la vista home del modulo auth Debe de renderizar correctamente el componente View 1`] = `
"<h1>Home Sidebar</h1>
<button class=\\"btn btn-primary\\">Primary</button>"
`;
